lib/index.js

```diff
@@ -15,6 +15,20 @@ function extend() {
     return result;
 }
 
+function debounce (func, wait, immediate) {
```

Comment:
It might be better to use the lodash debounce module. Same goes for the extend function, but that's outside the scope of this PR